pkg/cluster/provider.go

```diff
@@ -169,7 +169,7 @@ func ProviderWithPodman() ProviderOption {
 }
 
 // Create provisions and starts a kubernetes-in-docker cluster
-// TODO: move name to an option to override config
+// TODO: Replace by CreateWithOpts in the next major version
```

Comment:
We don't have major releases yet, kind is still in alpha.

We should be striving to avoid this though, we should probably just remove this TODO instead.

`name` is the primary key for the cluster, and API callers really ought to be keeping track of a `name` so they can handle deletion etc., I think we should just drop this TODO, avoid breaking callers, and acknowledge that all cluster calls must supply name.